### ISSUE TITLE
False Positive Unused Import of Class only Used for Type Hints

### ISSUE THREAD

**Issue by alexanderkiel** [borkdude/clj-kondo#911]
**version**

2020.05.09

**macro usage**

no

**platform**

JVM

**problem**

A class imported only for the use in type hints inside protocol definitions is reported as unused.

**repro**

```clojure
(ns repro
  (:import
    [clojure.lang IReduceInit]))

(defprotocol Db
  (-list-resources ^IReduceInit [db type start-id]))
```

```
$ clj-kondo --lint repro.clj 
repro.clj:3:19: warning: Unused import IReduceInit
linting took 34ms, errors: 0, warnings: 1
```

**expected behavior**

Not reporting the import as unused.

**Comment by alexanderkiel** [comment]
Works, thanks!
